src/ai/defaultai.h

```diff
@@ -375,10 +380,16 @@ struct DefaultAI : ComputerPlayer {
 	// id of iron as resource to identify iron mines in mines_per_type map
 	int32_t iron_resource_id = Widelands::INVALID_INDEX;
 
+	// Record with immovable attribute is created or collected by which building names and attributes
+	std::map<Widelands::MapObjectDescr::AttributeIndex, std::set<ImmovableAttribute>>
+	   buildings_immovable_attributes_;
+
 	// this is a bunch of patterns that have to identify weapons and armors for input queues of
 	// trainingsites
-	std::vector<std::string> const armors_and_weapons = {
-	   "ax", "armor", "helm", "lance", "trident", "tabard", "shield", "mask", "spear"};
+	// TODO(GunChleoc): Get rid of this hard-coding
+	std::vector<std::string> const armors_and_weapons = {"ax",     "armor",  "boots",   "helm",
```

Comment:
until we get rid of this we should incorporate the keywords sword and garment to deal with frisians Keyword lance can be deleted though